examples/dxc/src/main.rs

```diff
@@ -0,0 +1,226 @@
+use bindings::Windows::Win32::{
+    Globalization::CP_UTF8,
+    Graphics::Hlsl::*,
+    System::{
+        Diagnostics::Debug::ERROR_FILE_NOT_FOUND,
+        SystemServices::{BOOL, PWSTR},
+    },
+};
+use libloading::{Library, Symbol};
+use std::path::Path;
+use std::rc::Rc;
+use windows::*;
+
+// Only exists in newer DXC headers
+const DXC_CP_UTF8: u32 = CP_UTF8;
+
+#[cfg(target_os = "windows")]
+fn dxcompiler_lib_name() -> &'static Path {
+    Path::new("dxcompiler.dll")
+}
+
+#[cfg(target_os = "linux")]
+fn dxcompiler_lib_name() -> &'static Path {
+    Path::new("./libdxcompiler.so")
+}
+
+#[cfg(target_os = "macos")]
+fn dxcompiler_lib_name() -> &'static Path {
+    Path::new("./libdxcompiler.dynlib")
+}
+
+fn blob_encoding_as_str(blob: &IDxcBlobEncoding) -> &str {
+    let mut known: BOOL = false.into();
+    let mut cp = 0;
+    unsafe { blob.GetEncoding(known.set_abi(), cp.set_abi()) }.unwrap();
+    assert!(bool::from(known));
+    assert_eq!(cp, DXC_CP_UTF8);
+    unsafe {
+        let slice = std::slice::from_raw_parts(
+            blob.GetBufferPointer() as *const u8,
+            blob.GetBufferSize() - 1,
+        );
+        std::str::from_utf8_unchecked(slice)
+    }
+}
+
+fn create_blob(library: &IDxcLibrary, data: &str) -> windows::Result<IDxcBlobEncoding> {
+    let mut blob = None;
+    unsafe {
+        library.CreateBlobWithEncodingFromPinned(
+            data.as_ptr() as *const _,
+            data.len() as u32,
+            DXC_CP_UTF8,
+            &mut blob,
+        )
+    }
+    .and_some(blob)
+}
+
+#[allow(non_snake_case)]
+fn main() -> windows::Result<()> {
+    let lib = unsafe { Library::new(dxcompiler_lib_name()) }.unwrap();
+    let create: Symbol<DxcCreateInstanceProc> = unsafe { lib.get(b"DxcCreateInstance\0") }.unwrap();
+    dbg!(&create);
+
+    // let mut compiler: Option<IDxcCompiler2> = None;
+    // let compiler = unsafe { create(&CLSID_DxcCompiler, &IDxcCompiler2::IID, compiler.set_abi()) }
+    //     .and_some(compiler)
+    //     .unwrap();
+    let compiler: IDxcCompiler2 = unsafe { DxcCreateInstanceProc(&create, &CLSID_DxcCompiler) }?;
+    let library: IDxcLibrary = unsafe { DxcCreateInstanceProc(&create, &CLSID_DxcLibrary) }?;
```

Comment:
#747 ^^

Reply:
The commented out code would disappear, in favour of a single call to `DxcCreateInstanceProc`.